src/components/navbar/navLinkSection/NavLinkSection.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Fragment } from 'react';
+import { Link } from 'react-router-dom';
+import classes from './NavLinkSection.module.scss';
+import PropTypes from 'prop-types';
+
+const NavLinkSection = ({ navLink, closeDropdownNav, toggleNewsModal }) => {
+    return (
+        <Fragment>
+            {navLink.map(({ display, path, external, modal }) => {
+                return path ? (
+                    <Link
+                        key={display}
+                        className={classes.link}
+                        to={path}
+                        onMouseDown={closeDropdownNav}
+                    >
+                        {display}
+                    </Link>
+                ) : external ? (
+                    <a
+                        key={display}
+                        className={classes.link}
+                        href={external}
+                        target='_blank'
+                        rel='noopener noreferrer'
+                        onMouseDown={closeDropdownNav}
+                    >
+                        {display}
+                    </a>
+                ) : modal ? (
```

Comment:
The modal value doesn't seem to be used

Reply:
I did this to be ready for the future pretty much if these objects change. Also trying to stick with the convention of whoever wrote the constants. Plus if there is a way to store the proper functions in the constants file.
Perhaps we can create a functions export file. I have really only done so with redux.

Reply:
Not married to this idea, and I can think about new ways of representing this conditional.

Reply:
Sorry in the words of Amy, for the slow kids at the back of the class can you add a little comment about what going on here.

Reply:
If it does what I think it does this seems like a good approach. Is the modal option not needed in order to open the newsletter or am I just being slow?

Reply:
TBH I think switching this to a switch case may be more appropriate.
I was just stacking conditionals in the render component. Technically I could remove the ": modal (" and the ": null;"
It would work the same. I do need the modal to have a different onClick or mousedown handler is all. I could maybe add a conditional inside the click handler, but I have never tried and it sounds like it would read kinda confusing, maybe not.
My thinking was: the /links and anchors need to be separate so the anchor can take us to a new page, the modal needs to be different for a function call to open the modal. Arron's suggestion of calling the modal function earlier made this much easier if I can put the function call here instead of passing it props.

Reply:
I don't know what our future links might entail so I assumed we may have more external links or modal openings or other keys we need to make in our constants.js objs

Reply:
I changed this to a switch case, let me know if you think I made it worse 🌵

Reply:
Yeah I much prefer that thank you!